connector/src/org/bdinetwork/connector/interceptors.clj

```diff
@@ -0,0 +1,32 @@
+;;; SPDX-FileCopyrightText: 2025 Jomco B.V.
+;;; SPDX-FileCopyrightText: 2025 Topsector Logistiek
+;;; SPDX-License-Identifier: AGPL-3.0-or-later
+
+(ns org.bdinetwork.connector.interceptors
+  (:require [clojure.tools.logging :as log]
+            [nl.jomco.http-status-codes :as http-status]
+            [org.bdinetwork.authentication.access-token :as access-token]
+            [org.bdinetwork.gateway.interceptors :refer [->interceptor interceptor]]))
+
+(defn extract-client-id [request config]
+  (let [auth (get-in request [:headers "authorization"])]
+    (when-let [[_ token] (and auth (re-matches #"Bearer (\S+)" auth))]
+      (try
+        (access-token/access-token->client-id token config)
+        (catch Exception e
+          (log/info e "Invalid access token")
+          nil)))))
+
+(defmethod ->interceptor 'bdi/authenticate
```

Comment:
ergens documenteren dat je in de "niet-geauthenticeerde" flows eventuele binnenkomende "x-bdi-client-id" header moet wegfilteren!